astropy/coordinates/tests/test_distance.py

```diff
@@ -233,11 +232,11 @@ def test_parallax():
     with pytest.raises(ValueError):
         Distance(parallax=[10, 1, -1] * u.mas)
 
-    with catch_warnings(AstropyWarning) as w:
+    with pytest.warns(AstropyWarning) as w:
         Distance(parallax=-1 * u.mas, allow_negative=True)
     assert len(w) > 0
```

Comment:
No need to check the length (or assign to `w`), I would think

Reply:
Good catch!

Reply:
No need in fact to return it at all if you don't wanna analyse it.